tests/test_purger.py

```diff
@@ -0,0 +1,102 @@
+import json
+import string
+import time
+from random import randint
+from typing import Any, Dict
+
+import pytest
+
+from build.purger.purger import (USER_PURGING_THRESHOLD,
+                                 get_network_to_broadcast_rooms,
+                                 run_user_purger)
+from raiden.constants import Networks
+from tests.file_templates import USER_PRESENCE_TEMPLATE
+from tests.utils import TestGMatrixHttpApi, create_user_activity_dict
+
+
+@pytest.fixture
+def due_users(last_update_in_days: int, due_users_count: int) -> Dict[str, int]:
+    return create_user_activity_dict(
+        due_users_count,
+        USER_PURGING_THRESHOLD + 1,
+        USER_PURGING_THRESHOLD + last_update_in_days * 24 * 60 * 60,
+    )
+
+
+@pytest.fixture
+def active_users(active_users_count: int) -> Dict[str, int]:
+    return create_user_activity_dict(active_users_count, 0, USER_PURGING_THRESHOLD + 1)
+
+
+@pytest.fixture
+def global_user_activity(last_update_in_days, due_users, active_users, networks):
+
+    network_to_users_activity = dict()
+
+    for network in networks:
+        users = {**due_users, **active_users}
+        network_to_users_activity[network.value] = users
+
+    return json.loads(
+        string.Template(USER_PRESENCE_TEMPLATE).substitute(
+            last_update=last_update_in_days * 24 * 60 * 60,
+            network_to_users=json.dumps(network_to_users_activity),
+        )
+    )
+
+
+@pytest.fixture
+def activity_change(
+    due_users: Dict[str, int], activity_changed_count: Dict[str, int]
+) -> Dict[str, int]:
+
+    users = dict()
+    user_ids = list(due_users.keys())
+    activity_changed = min(len(due_users), activity_changed_count)
+    for _ in range(activity_changed):
+        user_id = user_ids.pop()
+        last_active_ago = randint(0, USER_PURGING_THRESHOLD + 1)
```

Comment:
Is `USER_PURGING_THRESHOLD + 1` right? I expected `USER_PURGING_THRESHOLD` or `USER_PURGING_THRESHOLD - 1`

Reply:
right

Reply:
Does this mean the current value is right, or that I am right with my expectation? It it's the latter, this still needs a change before merging or this will be flaky.